src/promptflow-evals/tests/evals/e2etests/test_evaluate.py

```diff
@@ -79,3 +85,25 @@ def test_evaluate_python_function(self, data_file):
         assert "answer.length" in metrics.keys()
         assert metrics.get("answer.length") == np.nanmean(row_result_df["outputs.answer.length"])
         assert row_result_df["outputs.answer.length"][2] == 31
+
+    def test_evaluate_with_target(self, questions_file):
+        """Test evaluation with target function."""
+        # We cannot define target in this file as pytest will load
```

Comment:
How about define the target function as an inner function inside test_evaluate_with_target? Would that work?

Reply:
That is correct, however in this case we will not be able to fully test the promptflow functionality. The issue is as follows:
1. pytest loads all modules at the beginning for optimization purposes (I guess?)
2. Now, when promptflow will try to import function_test, it will take it from already imported module (test_evaluate), but not from the saved flow, so if we will have issues importing flow, pyest will not catch it as the function is already imported.